torchgeo/datasets/geo.py

```diff
@@ -415,18 +416,24 @@ def __init__(
             transforms: a function/transform that takes an input sample
                 and returns a transformed version
             cache: if True, cache file handle to speed up repeated sampling
+            time_series: if True, return imagery as time series with shape [T,C,H,W]
+                and include 'dates' key with datetime information
```

Comment:
Let's not refer to it as "imagery", it may be masks or other raster data. Let's just call it "data". 

I'm not sure if "dates" is the right name for the key. It is a datetime, not a date. In other places in TorchGeo, we are using x, y, t for the centroids. We could change that to lon, lat, time.

Let's also add a description like:
"if False, return a merged mosaic with shape [C,H,W]"

Reply:
Renamed "dates" to "timestamps"